src/main/java/use_case/EventAdder.java

```diff
@@ -21,13 +22,11 @@
      *         true if the event is created and added to calendar successfully
      */
     public boolean addEvent(String eventName, LocalDateTime startTime,
-                            LocalDateTime endTime,
+                            LocalDateTime endTime, HashSet<String> tags, LocalDate date,
                             AccessCalendarData calendarData) {
-        Duration timeNeeded = Duration.between(startTime, endTime);
-        Task task = new Task(eventName, timeNeeded);
-        Event event = new Event(task, startTime, endTime.toLocalTime());
-        calendarData.addEvent(event);
-        return true;
+        Event event = new Event(eventName, startTime.toLocalTime(), endTime.toLocalTime(), tags, date);
+        calendarData.addEvent(event, event.getTask());
+        return true; // TODO: return value should indicate success of data creation
```

Comment:
I wasn't sure whether to return a boolean when creating tasks and events too. But Jenci and I decided to add this in when writing methods for the controller so that we can give some feedback to the user about whether the event/task has been created.

I am personally not sure what is the best practice and I think it is worth discussing how we should handle cases when event/task are not created successfully (use exception handling?)

So for phase 0 we can leave it like this and improve it after discussing. I placed a TODO there to remind ourselves about it.